webboot/wget_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name    string
+		isolink string
+		md5link string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			r, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
+			}
+
+			body, err := ioutil.ReadAll(r)
+			if err != nil {
+				t.Errorf("Failed to read the ISO: got %v, want nil", err)
+			}
+			//computes the hash of the ISO & converts it into a string to compare with md5.txt hash
+			md := md5.Sum(body)
+			hash := hex.EncodeToString(md[:])
+
+			c, err := wget(v.md5link)
+
+			var bytes []byte
+			if bytes, err = ioutil.ReadAll(c); err != nil {
+				t.Errorf("Failed to read md5.txt: got %v, want nil", err)
+			}
+			//checks multiple lines of md5.txt file and compares the hashes
+			var fields string
+			var found bool
+			for _, line := range strings.Split(string(bytes), "\n") {
+				if f := strings.Fields(line); len(f) != 0 && f[0] == hash {
+					fields = f[0]
+					found = true
+				}
+			}
+
+			if found != true {
```

Comment:
That is very true. Since fields would not arrive in the if statement if len(f) == 0 OR f[0] != hash, then fields would be an empty string, and there would be no point of returning an empty string to the tester. Instead, I got rid of initializing it on line 45 and using it on line 49 because the purpose of declaring the variable fields out of the for loop was to initialize it inside the for loop and use it outside afterwards to return to the test. But if it would be empty anyway and there was no where else i was using fields for then it could just be omitted. This saves time and space even if it makes the slightest impact. Thank you for pointing this out.